test/model/content_test.dart

```diff
@@ -292,19 +293,54 @@ void main() {
       QuotationNode([ParagraphNode(links: null, nodes: [TextNode('words')])]),
     ]);
 
-  testParse('parse code blocks, no language',
+  testParse('parse code blocks, without syntax highlighting',
     // "```\nverb\natim\n```"
     '<div class="codehilite"><pre><span></span><code>verb\natim\n</code></pre></div>', const [
-      CodeBlockNode(text: 'verb\natim'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'verb\natim', type: CodeBlockSpanType.text),
+      ]),
     ]);
 
-  testParse('parse code blocks, with highlighted language',
+  testParse('parse code blocks, with syntax highlighting',
     // "```dart\nclass A {}\n```"
     '<div class="codehilite" data-code-language="Dart"><pre>'
         '<span></span><code><span class="kd">class</span><span class="w"> </span>'
         '<span class="nc">A</span><span class="w"> </span><span class="p">{}</span>'
         '\n</code></pre></div>', const [
-      CodeBlockNode(text: 'class A {}'),
+      CodeBlockNode([
+        CodeBlockSpanNode(text: 'class', type: CodeBlockSpanType.keywordDeclaration),
+        CodeBlockSpanNode(text: ' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text: 'A', type: CodeBlockSpanType.nameClass),
+        CodeBlockSpanNode(text: ' ', type: CodeBlockSpanType.whitespace),
+        CodeBlockSpanNode(text: '{}', type: CodeBlockSpanType.punctuation),
+      ]),
+    ]);
+
+  testParse('parse code blocks, with syntax highlighting and highlighted lines',
+    // "```\n::markdown hl_lines="2 4"\n# he\n## llo\n### world\n```"
```

Comment:
Still not valid syntax :-) (continuing from https://github.com/zulip/zulip-flutter/pull/242#discussion_r1289874749 )

Try uncommenting the line, and you'll see parse errors. That's because there are double-quotes `"` inside the intended string, which interfere with the double-quotes used to enclose it.

I think the simplest fix is to enclose with single-quotes `'` instead, as in my previous suggestion. Triple quotes would also work:
```suggestion
    // """```\n::markdown hl_lines="2 4"\n# he\n## llo\n### world\n```"""
```

In principle another solution would be to escape the internal quotes, like `\"`. But I think that's less preferable because it makes it a bit harder to read — it takes a moment to work out that the backslashes aren't ultimately there in the string being represented.

Reply:
Ah of course, sorry I should've checked.